crypto/txscript/src/script_class.rs

```diff
@@ -44,13 +44,17 @@ const PUB_KEY_ECDSA: &str = "pubkeyecdsa";
 const SCRIPT_HASH: &str = "scripthash";
 
 impl ScriptClass {
-    pub fn from_script(script_public_key: &[u8]) -> Self {
-        if Self::is_pay_to_pubkey(script_public_key) {
-            ScriptClass::PubKey
-        } else if Self::is_pay_to_pubkey_ecdsa(script_public_key) {
-            Self::PubKeyECDSA
-        } else if Self::is_pay_to_script_hash(script_public_key) {
-            Self::ScriptHash
+    pub fn from_script(script_public_key: &[u8], version: ScriptPublicKeyVersion) -> Self {
```

Comment:
Maybe it should just get a ScriptPublicKey and separate it internally to script and version?

Reply:
given the use in go, makes sense, ok